search/django/weedid/views.py

```diff
@@ -119,13 +120,11 @@ def upload(request):
 
 
 class CustomUploader:
+    @require_http_methods(["POST"])
+    @login_required
     @classmethod
```

Comment:
Giving me this when I put @classmethod at the top

django            | Internal Server Error: /api/upload_voc/
django            | Traceback (most recent call last):
django            |   File "/usr/local/lib/python3.7/site-packages/django/core/handlers/exception.py", line 34, in inner
django            |     response = get_response(request)
django            |   File "/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py", line 115, in _get_response
django            |     response = self.process_exception_by_middleware(e, request)
django            |   File "/usr/local/lib/python3.7/site-packages/django/core/handlers/base.py", line 113, in _get_response
django            |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
django            |   File "/usr/local/lib/python3.7/site-packages/django/views/decorators/http.py", line 32, in inner
django            |     if request.method not in request_method_list:
django            | AttributeError: type object 'VocUploader' has no attribute 'method'

Reply:
https://stackoverflow.com/questions/31633259/django-how-to-use-decorator-in-class-based-view-methods

Above is not working as I tried. I reckon it's because the classmethod has cls as the first arg while the decorator is assuming request should be the first.